protocolupgrade/engine.go

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package protocolupgrade
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+
+	vegapb "code.vegaprotocol.io/protos/vega"
+	commandspb "code.vegaprotocol.io/protos/vega/commands/v1"
+	eventspb "code.vegaprotocol.io/protos/vega/events/v1"
+	"code.vegaprotocol.io/vega/events"
+	"code.vegaprotocol.io/vega/logging"
+	"code.vegaprotocol.io/vega/txn"
+	"code.vegaprotocol.io/vega/types"
+	"code.vegaprotocol.io/vega/types/num"
+	"github.com/blang/semver"
+	"github.com/cenkalti/backoff"
+	"github.com/golang/protobuf/proto"
+)
+
+var requiredQuorum = num.NewDecimalFromFloat(2).Div(num.NewDecimalFromFloat(3))
```

Comment:
Shouldn't we use the network parameters instead for this?

Reply:
also, I think each governance proposal type has its own so we can add a new one

Reply:
ok, I'll add a network parameter for it.